llvm_passes/core/FaultInjectionPass.cpp

```diff
@@ -93,6 +93,15 @@ void FaultInjectionPass::insertInjectionFuncCall(
       else fi_reg = fi_inst->getOperand(*reg_pos_it);
       //if(isa<Constant>(fi_reg))  continue;
       Type *returntype = fi_reg->getType();
+      
+      // The return type is not a valid return type, and should hence be ignored
+      // This is to deal with types such as Metadata that are not valid return types
+      // Or if the instruction is an intrinsic one (starts with 'llvm_', ignore it
+      if ( !FunctionType::isValidReturnType(returntype) ||
+		 isa<IntrinsicInst>(fi_inst) ) 
```

Comment:
Please align the `isa<IntrinsicInst>(fi_inst)` with the same indent as `!FunctionType`. Otherwise, everything else looks good.